oci/auth/aws/auth.go

```diff
@@ -33,15 +33,19 @@ import (
 	"github.com/fluxcd/pkg/oci"
 )
 
-var registryPartRe = regexp.MustCompile(`([0-9+]*).dkr.ecr.([^/.]*)\.(amazonaws\.com[.cn]*)/([^:]+):?(.*)`)
+var fullImagePathRe = regexp.MustCompile(`([0-9+]*).dkr.ecr.([^/.]*)\.(amazonaws\.com[.cn]*)/([^:]+):?(.*)`)
+var registryPartRe = regexp.MustCompile(`([0-9+]*).dkr.ecr.([^/.]*)\.(amazonaws\.com[.cn]*)`)
 
 // ParseImage returns the AWS account ID and region and `true` if
 // the image repository is hosted in AWS's Elastic Container Registry,
 // otherwise empty strings and `false`.
 func ParseImage(image string) (accountId, awsEcrRegion string, ok bool) {
-	registryParts := registryPartRe.FindAllStringSubmatch(image, -1)
+	registryParts := fullImagePathRe.FindAllStringSubmatch(image, -1)
 	if len(registryParts) < 1 || len(registryParts[0]) < 3 {
-		return "", "", false
+		registryParts = registryPartRe.FindAllStringSubmatch(image, -1)
+		if len(registryParts) < 1 || len(registryParts[0]) < 3 {
+			return "", "", false
+		}
```

Comment:
I went on to find some background on why this function parses image and not registry like it's done for GCP and Azure.
The image parsing and the regex was introduced in https://github.com/fluxcd/image-reflector-controller/pull/174 , which later was refactored into a package and moved to this repository.
Based on the above PR, I don't see any specific reason to have it different from the other providers. For GCP and Azure, only the registry host is checked to be valid. We should be able to do something similar for AWS.
I think the reason for the name to be specific to image is that it was added in image-reflector-controller, where the input is expected to be an image. Now that this is becoming more generic, we should be able to relax the parsing and make it about registry only. The information that's extracted after parsing are the account ID and region, which are still present in a registry URL.
Based on the above reasoning, I think we should rename the function to `ParseRegistry()`, then replace the old regex with the new ECR registry pattern, update login.go to use `aws.ParseRegistry()` and anywhere else where it's used, and update the old tests accordingly.
I think this change would be backwards compatible because we are relaxing the parser, shouldn't break any existing behavior.

Reply:
Thanks for digging up that background, @darkowlzz . I went ahead and tried refactoring `ParseImage` into `ParseRegistry` and relaxed the parsing. I believe I've gotten all the pieces you've mentioned and the updated tests are passing locally. Would you give it another look?

Reply:
Thanks for updating. The changes look good to me. I'll run the integration tests with cloud providers to make sure it still works.

Reply:
I ran the integration tests in this repo and in image-reflector-controller against AWS with these changes and they continue to work.

Reply:
We are about to do a release today. So, I'll just rebase and squash the commits myself. Hope that's okay. The new source-controller in this release will have this fix.

Reply:
Thanks!